kernel/Kconfig

```diff
@@ -951,6 +951,14 @@ config HAS_DYNAMIC_DEVICE_HANDLES
 	  Hidden option that makes possible to manipulate device handles at
 	  runtime.
 
+config DEVICE_STORE_NAME
```

Comment:
We need to figure out what this means for the SHELL if CONFIG_DEVICE_STORE_NAME=n

Reply:
Should `device_get_binding()` even exist if `CONFIG_DEVICE_STORE_NAME=n`, should `device_get_name`?

Reply:
I think they need to exist so that we don't require code changes to enable/disable it. device_get_binding() could just return NULL in that case, though.

Reply:
why, if you are setting this option, you clearly are concerned about space, so why have the added space taken up in the system?  Seems like we should `ifdef` out the name field in `struct device`.

Reply:
this PR already ifdeffes out `struct device` name field. We can simplify `device_get_binding` to return NULL if preferred.

Reply:
Seems my above concern is already discussed here; I agree with @galak on this one, at least for the default.